src/use-swr.ts

```diff
@@ -46,7 +46,12 @@ const MUTATION_TS = {}
 const MUTATION_END_TS = {}
 
 // setup DOM events listeners for `focus` and `reconnect` actions
-if (!IS_SERVER && window.addEventListener) {
+setup({
```

Comment:
this call is executed at the very beginning of the code. so if we want to customize this to support RN. the default config should be able to change at the begining. but I'm not sure if it's a good way to expose defaultConfig out

Reply:
Yeah, I think we can create 2 files which tackle this process:

```
src/
  use-swr.ts
  setup.ts
  setup.native.ts
```

And then inside `use-swr.js`, we simply `import './setup'`. IIRC RN will automatically import the native one.

(I never tested, need to verify)

Reply:
or could it become a preset?

```js
// on user land code
function rnPreset() {
   return {
     setOnFocus() {},
     isDocumentVisible() {}, 
     // ...
   }
}

SWRConfig.use(rnPreset()) // override the default config
```

since we renamed the context provider to `SWRConfig` for convenience, I'm wondering if it's possible to add more special methdos onto it. just a thought.

Reply:
How about:

```jsx
<SWRConfig preset={[rn]} use={[...plugins]} config={cfg} />
```

Reply:
I feel provider doesn't solve this since currently stepup process is happened at the beginning without provider specified, which is before the hook gets render.

if we setup in the provider, will we still keep the default setup for web browser runtime?